src/core/tsi/ssl_transport_security.h

```diff
@@ -53,6 +54,9 @@
 
 using RootCertInfo = std::variant<std::string, grpc_core::SpiffeBundleMap>;
 
+using PrivateKey =
+    std::variant<absl::string_view, grpc_core::CustomPrivateKeySign>;
```

Comment:
The TSI library should not depend on code in the `grpc_core` namespace; the dependencies need to go in the opposite direction.  This is because the TSI library should be usable in projects other than gRPC that may not want to depend on other gRPC code.

I suggest moving the definition of `CustomPrivateKeySign` into the TSI library and then referring to it from the gRPC code.